File: frontend/mela/lib/domain/entity/lecture/lecture.dart
class Lecture {
  final String lectureId;
  final String levelId;
  final String topicId;
  final String lectureName;
  final int ordinalNumber;
  final String lectureDescription;
  final int totalExercises;
  final int totalPassExercises;

  Lecture({
    required this.lectureId,
    required this.levelId,
    required this.topicId,
    required this.lectureName,
    required this.ordinalNumber,
    required this.lectureDescription,
    required this.totalExercises,
    required this.totalPassExercises,
  });
  double get progress => totalPassExercises / totalExercises;

  factory Lecture.fromJson(Map<String, dynamic> json) {
    return Lecture(
      lectureId: json['lectureId'],
      levelId: json['levelId'],
      topicId: json['topicId'],
      lectureName: json['name'],
      ordinalNumber: json['ordinalNumber'] ?? 0,
      lectureDescription: json['description'],
      totalExercises: json['totalExercises'],
      totalPassExercises: json['totalPassExercises'],
    );
  }
}

File: backend/mela/src/main/java/com/hcmus/mela/lecture/model/Lecture.java
package com.hcmus.mela.lecture.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;
import org.springframework.data.annotation.Id;
import org.springframework.data.annotation.Transient;
import org.springframework.data.mongodb.core.mapping.Document;
import org.springframework.data.mongodb.core.mapping.Field;

import java.util.List;
import java.util.UUID;

@Data
@NoArgsConstructor
@AllArgsConstructor
@ToString
@Document(collection = "lectures")
public class Lecture {

    @Id
    private UUID lectureId;

    @Field("level_id")
    private UUID levelId;

    @Field("topic_id")
    private UUID topicId;

    @Field("name")
    private String name;

    @Field("ordinal_number")
    private Integer ordinalNumber;

    @Field("description")
    private String description;

    @Field("sections")
    private List<Section> sections;

    @Field("total_exercises")
    private Integer totalExercises;
}
